### PR TITLE
Controllers : Adding a moving average for the velocity field returned by the Hydra controller

### PR DESCRIPTION
**Pull request by samcake**
Hand a js script to represent the HandPose of MyAvatar
 and the velocity vector in real time

- added the necessary to also filter the angularVelocity but not used yet because that suppose to do it not in the quaternion space but using a different mathematical object (Tony knows!)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d400c694f6d821f912002173b0c4097c54d32eaa available at: https://gist.github.com/8d0946a7f32c2bf569df<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eae316100a91193b107a7d4ca9c2a874cbc186f1 available at: https://gist.github.com/8d0946a7f32c2bf569df<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.dmg

File: libraries/input-plugins/src/input-plugins/SixenseManager.h [hunk 1]
```diff
@@ -94,6 +94,37 @@ public slots:
     glm::vec3 _reachRight;
     float _lastDistance;
     bool _useSixenseFilter = true;
+
+    template <class T, int MAX_NUM_SAMPLES> class MovingAverage {
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: we definitely need to put this in `libraries/shared` - and we should probably replace the current float based SimpleMovingAverage with an instance of this

File: libraries/input-plugins/src/input-plugins/SixenseManager.h [hunk 2]
```diff
@@ -94,6 +94,37 @@ public slots:
...
+        }
+
+        bool isAverageValid() const { return !samples.empty(); }
+
+        void addSample(T sample) {
+            samples.push_front(sample);
+            int numSamples = samples.size();
+
+            if (numSamples < MAX_NUM_SAMPLES) {
+                average = (sample + average * float(numSamples - 1)) / float(numSamples);
```
Comment (ZappoMan) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: coding standard - 3.1.2.1 Primitive types should use C style casting

File: libraries/input-plugins/src/input-plugins/SixenseManager.h (hunk 1 above)
Comment (samcake) [inline libraries/input-plugins/src/input-plugins/SixenseManager.h]: It s slightly different as it is not based on time but just number of smaple bu t i ll move it there next to the other one

**Comment by hifi-gustavo** [comment]
Build console output for commit: a8872d065b2b4aa680b49b86b99680b43e8c31a0 available at: https://gist.github.com/8d0946a7f32c2bf569df<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.dmg

File: libraries/shared/src/SimpleMovingAverage.h [hunk 3]
```diff
@@ -40,4 +41,31 @@ class SimpleMovingAverage {
     float ONE_MINUS_WEIGHTING;
 };
 
+
+template <class T, int MAX_NUM_SAMPLES> class MovingAverage {
+public:
+    using Samples = std::list< T >;
```
Comment (ZappoMan) [inline libraries/shared/src/SimpleMovingAverage.h]: why do you need to keep a list? can you implement this using the same approach as SimpleMovingAverage?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 990764d85577ed5f9e167b3fab91bbb6f1176407 available at: https://gist.github.com/8d0946a7f32c2bf569df<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6174.dmg

**Pull request merged**